### DISCUSSION TITLE
about pnpm import yarn.lock in monorepo

### DISCUSSION THREAD

**Discussion by foora** [pnpm/pnpm#3798] [Q&A]
I have a monorepo which use **yarn workspace** + **lerna**. 
Now, I want switch to **pnpm**!  
But when I use **pnpm import** to transform **yarn.lock** to **pnpm-lock.yaml**.
I found it just transform the dependencies which i define in the root **package.json** and ignore the dependencies which i define in the workspace's packages.
So, I have to merge all package.json in my workspace to the root. and then run **pnpm import** to resolve this problem.
Will you have an idea to let **pnpm import** support monorepo?

**Comment by zkochan** [comment]
Don't you have a single `yarn.lock` in the root of the monorepo? I think it should work correctly in that case. In this pull request was the feature added: https://github.com/pnpm/pnpm/pull/3655

**Reply by foora** [reply]
I have a single `yarn.lock` in the root. but when i run `pnpm import`, it will ignore to find all the package.json  in the workspace's packages, and only find the package.json in the root of the monorepo.

https://github.com/pnpm/pnpm/blob/fe5688dc0c80385089f9abd95b466d54dae4aa56/packages/supi/src/install/index.ts#L134-L138
either `pnpm install` or `pnpm import` will call `mutateModules` to generate `pnpm-lock.yaml` file and pass an Array called `projects`. `pnpm install` will pass all package.json manifest. but `pnpm import` not. 

Is it the reseaon that `pnpm import` work incorrectly in workspace?

**Reply by zkochan** [reply]
if it works as you describe then it looks like a bug
